src/java.base/share/classes/java/text/StringBufFactory.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2024, Alibaba Group Holding Limited. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.text;
+
+/**
+ * StringBufFactory create {code Format.StringBuf}'s implements that
+ * backend with {code StringBuffer} and {code StringBuilder}.
+ * It used by {code Format}'s implements to replace inner string
+ * buffer from {code StringBuffer} to {code StringBuilder} to gain
+ * a better performance.
+ */
+final class StringBufFactory {
+
+    static Format.StringBuf of(StringBuffer sb) {
```

Comment:
At least for the factory class, let's just `import java.text.Format.StringBuf;`, so we don't have to use the full name everywhere.